src/java/org/apache/cassandra/index/sai/disk/v1/SegmentBuilder.java

```diff
@@ -393,11 +394,9 @@ protected long addInternalAsync(ByteBuffer term, int segmentRowId)
         @Override
         protected SegmentMetadata.ComponentMetadataMap flushInternal() throws IOException
         {
-            // VSTODO this is a bit of a hack. We call computeDeletedOrdinals to call computeRowIds on each
-            // VectorPostings, which will populate the rowIds field, but if we refactor the code, we could skip that.
```

Comment:
This logic could still be improved. Are you removing this comment because this code path is only hit when `ENABLE_LTM_CONSTRUCTION` is not enabled, and `ENABLE_LTM_CONSTRUCTION` will be enabled all of the time eventually?

Reply:
what do you have in mind?  we do need a "should we actually write index components" method, and it depends on information that isn't safe to compute while it's being modified

Reply:
I'll change it to this:
```
            var shouldFlush = graphIndex.preFlush(p -> p);
            // there are no deletes to worry about when building the index during compaction,
            // and SegmentBuilder::flush checks for the empty index case before calling flushInternal
            assert shouldFlush;
```

I don't thing the memory savings (on compaction nodes, not workers) are worth adding another code path to maintain, especially since as you note the on-heap path will not be as common soon.  Happy to revisit if that proves wrong.